src/impact/capability-representation.md

```diff
@@ -0,0 +1,92 @@
+## Capability representation in memory
+
+Underlying implementations of CHERI are diverse, spanning 32-bit
+microcontrollers (such as CHERIoT) to 64-bit server-class processors (such as
+Arm's Morello).
+CHERI C/C++ provide broad flexibility for implementations to represent
+capability metadata in ways most suitable to their individual requirements.
+One specific area in which CHERI implementations may differ is in the specific
+in-memory representations of capabilities, due to not just different address
+sizes, but also different tradeoffs around bounds compression, permissions,
+and so on.
+
+CHERI C/C++ in general expect that capabilities will be accessed via pointer
+types, with operations such as dereferencing a pointer or performing pointer
+arithmetic implemented by compiler-generated code.
+Broadly, a capability consists of three parts: An address, inline metadata,
+and a validity tag.
+When stored in memory, CHERI capabilities consist of two adjacent
+address-sized words, stored at capability alignment:
+
+ - The lower address-sized word holds the address.
+ - The higher address-sized word holds the inline metadata.
+
+The capability validity tag is not directly addressable in memory.
+
+### In-memory representation of the capability address
+
+The address word is stored in the native integer representation of the
+architecture, and will be identical to the value returned by
+[`cheri_address_get()`](../apis/retrieving-capability-properties.md) for the
+capability.
+Taking a pointer to an in-memory capability and casting it to an address-sized
```

Comment:
This isn’t true of big-endian CHERI-MIPS, and I would be cautious to guarantee it as a result. In general using a narrower type on big-endian architectures for your load doesn’t work, this wasn’t just an oddity of CHERI, it was in fact consistent with how aliasing integers works.